sycl/include/CL/sycl/atomic.hpp

```diff
@@ -197,6 +198,37 @@ class atomic {
                   "T and pointerT must be same size");
   }
 
+#ifdef __SYCL_DEVICE_ONLY__
+  AtomicPtrType *getPtr() const { return Ptr; }
```

Comment:
Shouldn't this getPtr() method be moved to private methods section?

Reply:
Unfortunately not. In this case during implicit conversion from below we couldn't `getPtr` from `RHS` to cast it to `Ptr` from `this`.

Reply:
Also think getPtr should not be available to the user. You can "friend" classes to be able to accesses private methods.

Reply:
Done